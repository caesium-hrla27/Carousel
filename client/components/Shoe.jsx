import React, { Component } from 'react';
import styled from 'styled-components';
import Reviews from './Reviews.jsx';


import { Li, Text, Color, ShoeDiv, Prices, OrigPrice } from './style.js';

class Shoe extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isHovering: false
    };
    this.handleMouseEnter = this.handleMouseEnter.bind(this);
    this.handleMouseLeave = this.handleMouseLeave.bind(this);
  }

  handleMouseEnter(reviewsNum) {
    if (reviewsNum) {
      this.setState({
        isHovering: true
      });
    }
  }  

  handleMouseLeave() {
    this.setState({
      isHovering: false
    });
  }

  render() {
    
    return (

      <Li order={this.props.order} onMouseEnter={() => this.handleMouseEnter(this.props.reviewsNum)} onMouseLeave={() => this.handleMouseLeave()}>
      
        <img src={this.props.shoeUrl} /><br />
      
        <Text>

          <ShoeDiv>
            {this.props.reviewsNum && this.state.isHovering ? <Reviews reviewsNum={this.props.reviewsNum} reviewsAvg={this.props.reviewsAvg} /> : 
              <Color>{this.props.colors} {this.props.colors === 1 ? 'Color' : 'Colors'}<br /></Color>} 
            {this.props.name}<br />
          </ShoeDiv>
          {this.props.category === 'men_athletic' ? 'Basketball Shoe' : this.props.category}<br />
        
          <Prices>
            <OrigPrice origPrice={this.props.price} salePrice={this.props.salePrice}>
              {`$${this.props.price}`}
            </OrigPrice>
            {this.props.salePrice ? `$${this.props.salePrice}` : null}<br />
          </Prices>

        </Text>
      </Li>

    );

  }
}

export default Shoe;