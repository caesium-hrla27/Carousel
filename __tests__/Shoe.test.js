import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import expect from 'expect';
import toJson from 'enzyme-to-json';

// Component to be tested
import Shoe from '../client/components/Shoe';

// Configure Enzyme to work with Jest

Enzyme.configure({ adapter: new Adapter() });

const shallow = Enzyme.shallow;

describe('Shoe', () => {
  let shoe;

  beforeEach(() => {
    shoe = shallow(<Shoe />);  
  });

  it('should render properly', () => {
    const component = shoe.dive();

    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render an li', () => {
    expect(shoe.find('#li').length).toEqual(1);
  });

  it('should render an image', () => {
    expect(shoe.find('img').length).toEqual(1);
  });

  it('should render shoeText', () => {
    expect(shoe.find('#shoeText').length).toEqual(1);
  });

  it('should render shoePrices', () => {
    expect(shoe.find('#shoePrices').length).toEqual(1);
  });



  // beforeEach(() => {
  //   fullCarousel = mount(<Carousel />);
  // });

  // it('should render dynamically generated child components', () => {
  //   expect(fullCarousel.find('Shoe').length).toEqual(12);
  // });



});