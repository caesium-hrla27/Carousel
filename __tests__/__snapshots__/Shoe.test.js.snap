// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shoe should render properly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-gzVnrw",
        "isStatic": false,
        "rules": Array [
          "
  flex: 1 0 30%;
  margin-right: 20px;
  order: ",
          [Function],
          ";
  color: rgb(109, 109, 109);
  ",
        ],
      },
      "displayName": "styled.li",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-gzVnrw",
      "target": "li",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  id="li"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <img />
  <br />
  <styled.div
    id="text"
  >
    <styled.div
      id="shoeText"
    >
      <styled.div>
         
        Colors
        <br />
      </styled.div>
      <br />
    </styled.div>
    <br />
    <styled.div
      id="shoePrices"
    >
      <styled.div>
        $undefined
      </styled.div>
      <br />
    </styled.div>
  </styled.div>
</StyledComponent>
`;
